src/main/java/net/snowflake/client/jdbc/SnowflakeChunkDownloader.java

```diff
@@ -663,7 +675,8 @@ private void waitForChunkReady(SnowflakeResultChunk currentChunk) throws Interru
                     chunkHeadersMap,
                     networkTimeoutInMilli,
                     session));
-        downloaderFutures.put(nextChunkToDownload, downloaderFuture);
+        downloaderFutures.put(nextChunkToConsume, downloaderFuture);
+        nextChunkToDownload = nextChunkToConsume + 1;
```

Comment:
How did this ever work without this incrementing?

Reply:
I think previously we don't have test that cover this...